client/transport/jsonrpc/src/test/java/io/a2a/client/transport/jsonrpc/JsonMessages.java

```diff
@@ -288,110 +302,104 @@ public class JsonMessages {
-                     "mimeType": "text/plain"
+              "jsonrpc":"2.0",
+              "result":{
+                "id":"de38c76d-d54c-436c-8b9f-4c2703648d64",
+                "contextId":"c295ea44-7543-4f78-b524-7a38915ad6e4",
+                "status":{
+                  "state":"TASK_STATE_COMPLETED"
+                },
+                "artifacts":[
+                  {
+                    "artifactId":"artifact-1",
+                    "parts":[
+                      {
+                        "text":"Why did the chicken cross the road? To get to the other side!"
+                      }
+                    ]
                   }
-                 },
-                 {
-                  "kind": "file",
-                  "file": {
-                     "bytes": "aGVsbG8=",
-                     "name": "hello.txt"
+                ],
+                "history":[
+                  {
+                    "role":"ROLE_USER",
+                    "parts":[
+                      {
+                        "text":"tell me a joke"
+                      },
+                      {
+                        "file":{
+                          "file_with_uri":"file:///path/to/file.txt",
+                          "mediaType":"text/plain"
+                        }
+                      },
+                      {
+                        "file":{
+                          "file_with_bytes":"aGVsbG8=",
+                          "name":"hello.txt"
+                        }
+                      }
+                    ],
+                    "messageId":"message-123"
                   }
-                 }
                 ],
-                "messageId": "message-123",
-                "kind": "message"
-               }
-              ],
-              "metadata": {},
-              "kind": "task"
-             }
+                "metadata":{
+                  
+                }
+              }
             }
             """;
 
     static final String CANCEL_TASK_TEST_REQUEST = """
             {
-             "jsonrpc": "2.0",
-             "method": "tasks/cancel",
-             "params": {
-              "id": "de38c76d-d54c-436c-8b9f-4c2703648d64",
-              "metadata": {}
-             }
+              "jsonrpc":"2.0",
+              "method":"CancelTask",
+              "params":{
+                "name":"tasks/de38c76d-d54c-436c-8b9f-4c2703648d64"
+              }
             }
             """;
 
     static final String CANCEL_TASK_TEST_RESPONSE = """
             {
-             "jsonrpc": "2.0",
-             "result": {
-              "id": "de38c76d-d54c-436c-8b9f-4c2703648d64",
-              "contextId": "c295ea44-7543-4f78-b524-7a38915ad6e4",
-              "status": {
-               "state": "canceled"
-              },
-              "metadata": {},
-              "kind" : "task"
-             }
+              "jsonrpc":"2.0",
+              "result":{
```

Comment:
idem (I'd be expecting a `task` field)

Reply:
I agree with you that the spec does not clearly specify the returned payload but this would be consistent with https://a2a-protocol.org/latest/specification/#941-sendmessage